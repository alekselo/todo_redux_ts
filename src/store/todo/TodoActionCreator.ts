import { generateId } from "../../helpers/helper";

export const addTodo = (text: string) => {
  return {
    type: 'ADD' as const,
    payload: {id: generateId(), text, completed: false},
  }
};
 
export const deleteTodo = () => {
  return {
    type: 'DELETE' as const,
    payload: '',
  }
};
 
export const editTodo = () => {
  return {
    type: 'EDIT' as const,
    payload: {},
  }
};
 
export const completeTodo = () => {
  return {
    type: 'COMPLETE' as const,
    payload: '',
  }
};