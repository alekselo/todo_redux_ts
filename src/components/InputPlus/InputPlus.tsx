
import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';


import * as actions from '../../store/todo/TodoActionCreator';

import styles from './InputPlus.module.scss';

export const InputPlus = () => {

  const dispatch = useDispatch();
  useSelector(state => state);


  const [text, setText] = useState('');

  const onClick = () => {
    if (text.trim() !== '') {
      dispatch(actions.addTodo(text.trim()))
    }
    setText('');
  }

  return (<div className={styles.inputPlus}>
    <input
      value={text}
      onChange={(e) => { setText(e.target.value) }}
      type="text"
      className={styles.inputPlusValue}
      placeholder="Type here..."
      onKeyDown={(evt) => {
        if (evt.key === 'Enter') {
          onClick();
        }
      }}
    />
    <button
      type='submit'
      aria-label="Add"
      className={styles.inputPlusButton}
      onClick={onClick}
      disabled={!text}
    />
  </div>)

};

