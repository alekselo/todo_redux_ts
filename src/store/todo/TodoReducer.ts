
import * as actionCreators from './TodoActionCreator'

type Actions = typeof actionCreators[keyof typeof actionCreators];

export type StateType = {
  id: string,
  text: string,
  completed: boolean,
};

const initialTodoState: Array<StateType> = [
    { id: '0', text: 'Learn React', completed: true },
    { id: '1', text: 'Learn Redux', completed: false },
    { id: '2', text: 'Build something fun!', completed: false },
]



export const reducer = (state = initialTodoState, action: ReturnType<Actions>): Array<StateType> => {
  switch (action.type) {
    case 'ADD':
      return [...state, {id: action.payload.id, text: action.payload.text, completed: action.payload.completed }]
    default:
      return state;
    
  }

} 