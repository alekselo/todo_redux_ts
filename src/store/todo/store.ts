import { createStore } from 'redux';
import { reducer } from './TodoReducer';


export const store = createStore(reducer);

