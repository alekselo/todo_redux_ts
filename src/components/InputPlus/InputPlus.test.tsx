import { InputPlus } from "./InputPlus";
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../store/todo/store';

it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><InputPlus /></Provider>).toJSON();

  expect(tree).toMatchSnapshot();
});