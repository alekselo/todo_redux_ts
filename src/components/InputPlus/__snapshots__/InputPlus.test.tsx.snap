// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Type here..."
    type="text"
    value=""
  />
  <button
    aria-label="Add"
    disabled={true}
    onClick={[Function]}
    type="submit"
  />
</div>
`;
